climada/hazard/centroids/centr.py

```diff
@@ -79,280 +52,215 @@ class Centroids():
+        """ Return latitudes """
+        return self.gdf.geometry.y.values
+
+    @property
+    def lon(self):
+        """ Return longitudes """
+        return self.gdf.geometry.x.values
+
+    @property
+    def geometry(self):
+        """ Return the geometry """
+        return self.gdf['geometry']
+
+    @property
+    def on_land(self):
+        """ Get the on_land property """
+        if self.gdf.on_land.isna().all():
+            return None
+        return self.gdf['on_land'].values
+
+    @property
+    def region_id(self):
+        """ Get the assigned region_id """
+        if self.gdf.region_id.isna().all():
+            return None
+        return self.gdf['region_id'].values
+
+    @property
+    def crs(self):
+        """ Get the crs"""
+        return self.gdf.crs
+
+    @property
+    def size(self):
+        """Get size (number of lat/lon paris)"""
+        return self.gdf.shape[0]
+
+    @property
+    def shape(self):
+        """Get shape [lat, lon] assuming rastered data."""
+        return (np.unique(self.lat).size, np.unique(self.lon).size)
+
+    @property
+    def total_bounds(self):
+        """Get total bounds (minx, miny, maxx, maxy)."""
+        return self.gdf.total_bounds
+
+    @property
+    def coord(self):
+        """Get [lat, lon] array."""
+        return np.stack([self.lat, self.lon], axis=1)
+
+    def __eq__(self, other):
         """Return True if two centroids equal, False otherwise
 
         Parameters
         ----------
-        centr : Centroids
+        other : Centroids
             centroids to compare
 
         Returns
         -------
         eq : bool
         """
-        if self.meta and centr.meta:
-            return (u_coord.equal_crs(self.meta['crs'], centr.meta['crs'])
-                    and self.meta['height'] == centr.meta['height']
-                    and self.meta['width'] == centr.meta['width']
-                    and self.meta['transform'] == centr.meta['transform'])
-        return (u_coord.equal_crs(self.crs, centr.crs)
-                and self.lat.shape == centr.lat.shape
-                and self.lon.shape == centr.lon.shape
-                and np.allclose(self.lat, centr.lat)
-                and np.allclose(self.lon, centr.lon))
-
-    @staticmethod
-    def from_base_grid(land=False, res_as=360, base_file=None):
-        """Initialize from base grid data provided with CLIMADA
+        return self.gdf.equals(other.gdf) & u_coord.equal_crs(self.crs, other.crs)
 
-        Parameters
-        ----------
-        land : bool, optional
-            If True, restrict to grid points on land. Default: False.
-        res_as : int, optional
-            Base grid resolution in arc-seconds (one of 150, 360). Default: 360.
-        base_file : str, optional
-            If set, read this file instead of one provided with climada.
+    def to_default_crs(self):
+        """Project the current centroids to the default CRS (epsg4326)
+        Modifies the object in place.
```

Comment:
Remove this method in favor of setting the default CRS as default parameter of `to_crs`?

Reply:
Currently `to_crs` is thought as a user method and returns a new centroids object. On the other hand, `to_default_crs` is mostly there to handle internal methods that require the default crs. Could also be merged.

Reply:
Done in [e5dffd9](https://github.com/CLIMADA-project/climada_python/pull/787/commits/e5dffd97a6909e51173d3fceb20863bf7d740c4a)

Reply:
`centroids.to_default_crs(inplace=False)` currently results in a dead end - neither returning centroids nor setting the projection inplace.